### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by FFroehlich**
fixed SBML import

### CODE REVIEW

File: symbolic/am_and.m [hunk 1]
```diff
@@ -1,6 +1,11 @@
...
-fun = a*b;
+% am_and is the amici implementation of the symbolic and function
+%
+% Parameters:
+%  a: first input parameter @type sym
+%  b: second input parameter @type sym
+%
+% Return values:
+%  fun: logical value, negative for false, positive for true
+fun = am_min(a,b);
```
Comment (dweindl) [inline symbolic/am_and.m]: how is am_and the same as am_min?

File: symbolic/am_or.m [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
-% fun = f(a,b);
+% am_or is the amici implementation of the symbolic or function
+%
+% Parameters:
+%  a: first input parameter @type sym
+%  b: second input parameter @type sym
+%
+% Return values:
+%  fun: logical value, negative for false, positive for true
 fun = am_max(a,b);
```
Comment (dweindl) [inline symbolic/am_or.m]: how is "or" the same as "max"?

File: symbolic/am_and.m (hunk 1 above)
Comment (FFroehlich) [inline symbolic/am_and.m]: with the mapping

true -> positive
false -> negative

minimum  is only positive (true) if both a and b are positive (true)

File: symbolic/am_or.m (hunk 2 above)
Comment (FFroehlich) [inline symbolic/am_or.m]: with the mapping

true -> positive
false -> negative

maximum is only positive (true) if at least one of a and b is positive (true)

Comment (dweindl) [inline symbolic/am_or.m]: Got it.

Comment (dweindl) [inline symbolic/am_or.m]: That mapping is done within the symbolic toolbox?

Comment (FFroehlich) [inline symbolic/am_or.m]: that mapping is implicitely done in AMICI at runtime. In CVODES events are defined as roots with odd multiplicity. In SBML events always fire when triggers transition from false to true.

The advantage of the continuous mapping as that math is a little bit more rigorous and the "distance" to true/false can be used for regularisation of event observables that did not happen during simulation

Comment (dweindl) [inline symbolic/am_or.m]: Ah, k, thx

File: @SBMLode/SBMLode.m [hunk 3]
```diff
@@ -62,7 +64,7 @@
             model.checkODE();
         end
         
-        importSBML(this,filename)
+        importSBML(this,modelname)
```
Comment (FFroehlich) [inline @SBMLode/SBMLode.m]: you are correct, the argument should be filename and not modelname.

File: @SBMLode/importSBML.m [hunk 4]
```diff
@@ -457,42 +656,87 @@ function makeSubs(this,old,new)
...
+function z = delay(x,y)
+error('Events with delays are currently not supported!');
+end
+
+function x = stoich_initAssign_rule(y,initassignments_sym,initassignments_math,rulevars,rulemath)
+x = {};
+for iy = 1:length(y)
+    x{iy} = sym(y(iy).stoichiometry);
+    if(~isempty(sym(y(iy).id)))
+        if(~isempty(initassignments_sym) || ~isempty(rulevars))
```
Comment (FFroehlich) [inline @SBMLode/importSBML.m]: correct, fixed.

**Comment by FFroehlich** [comment]
valgrind says everything is fine, any objections on merge?

**Comment by paulstapor** [comment]
If amiexamples gives the same results as it does on the master, no.

**Comment by FFroehlich** [comment]
I checked and results for all examples look fine now (we should really set up regression tests for this).

**Pull request merged**